modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManagerTest.java

```diff
@@ -0,0 +1,944 @@
+                .thenReturn(getEncodeSCIMExceptionObject(new CharonException()));
+
+        SCIMResponse outputScimResponse = groupResourceManager.delete(id, userManager);
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), ResponseCodeConstants.CODE_INTERNAL_ERROR);
+
+    }
+
+    @DataProvider(name = "dataForTestUpdateGroupWithPUTSuccess")
+    public Object[][] dataToTestUpdateGroupWithPUTSuccess()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+
+        Group groupOld = decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING, schema, new Group());
+        String id = groupOld.getId();
+
+        Group groupNew = decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING_UPDATED, schema, new Group());
+
+        return new Object[][]{
+                {id, NEW_GROUP_SCIM_OBJECT_STRING_UPDATED, "id", null, groupNew, groupOld}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestUpdateGroupWithPUTSuccess")
+    public void testUpdateGroupWithPUTSuccess(String id, String scimObjectString, String
+            attributes, String excludeAttributes, Object objectNEWGroup, Object objectOLDGroup)
+            throws BadRequestException, NotFoundException, CharonException, NotImplementedException {
+
+        Group groupNew = (Group) objectNEWGroup;
+        Group groupOld = (Group) objectOLDGroup;
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_GROUP_ENDPOINT + "/" + GROUP_ID);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+
+        when(userManager.getGroup(id,
+                ResourceManagerUtil.getAllAttributeURIs(schema))).thenReturn(groupOld);
+
+        Group validatedGroup = (Group) ServerSideValidator.validateUpdatedSCIMObject(groupOld, groupNew, schema);
+        when(userManager.updateGroup(anyObject(), anyObject(), anyObject())).thenReturn(validatedGroup);
+
+        SCIMResponse outputScimResponse = groupResourceManager.updateWithPUT(id, scimObjectString, userManager,
+                attributes, excludeAttributes);
+        JSONObject obj = new JSONObject(outputScimResponse.getResponseMessage());
+
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), ResponseCodeConstants.CODE_OK);
+
+        String returnedURI = outputScimResponse.getHeaderParamMap().get("Location");
+        String expectedURI = "null/" + obj.getString("id");
+        Assert.assertEquals(returnedURI, expectedURI);
+
```

Comment:
Unnecessary new line.

Reply:
Fixed in commit 71483cb81fc57041180aaa8bc31aa99110211062